### PR TITLE
[DO NOT MERGE] Updated entities for incremental update

### PR DESCRIPTION
**Pull request by sthirugn**

### CODE REVIEW

**Comment by omaciel** [comment]
Can haz tests? :)

File: nailgun/entities.py [hunk 1]
```diff
@@ -1124,6 +1128,23 @@ def promote(self, synchronous=True, **kwargs):
         response = client.post(self.path('promote'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def inc_update(self, synchronous=True, **kwargs):
```
Comment (Ichimonji10) [inline nailgun/entities.py]: Can you rename this to `incremental_update`? I'm trying to keep method names identical to the underlying path called. I'm going to add a decorator at some point that'll formalize this convention.

File: nailgun/entities.py [hunk 2]
```diff
@@ -1105,6 +1105,10 @@ def path(self, which=None):
             return '{0}/promote'.format(
                 super(ContentViewVersion, self).path(which='self')
             )
+        if which == 'incremental_update':
```
Comment (Ichimonji10) [inline nailgun/entities.py]: Please mention this new path in the docstring.

File: nailgun/entities.py [hunk 3]
```diff
@@ -1598,12 +1619,24 @@ def create_payload(self):
         return {u'environment': super(Environment, self).create_payload()}
 
 
-class Errata(Entity):
+class Errata(Entity,
```
Comment (Ichimonji10) [inline nailgun/entities.py]: All on one line.

File: nailgun/entities.py (hunk 1 above)
Comment (Ichimonji10) [inline nailgun/entities.py]: Please add some test coverage for this method. You can just copy and paste the relevant bits from https://github.com/SatelliteQE/nailgun/compare/master...Ichimonji10:inc-update

File: nailgun/entities.py [hunk 4]
```diff
@@ -1105,6 +1105,10 @@ def path(self, which=None):
             return '{0}/promote'.format(
                 super(ContentViewVersion, self).path(which='self')
             )
+        if which == 'incremental_update':
+            return '{0}/incremental_update'.format(
+                super(ContentViewVersion, self).path(which=None)
+            )
```
Comment (abalakh) [inline nailgun/entities.py]: How about
```python
        if which in ('promote', 'incremental_update'):
            return '{0}/{1}'.format(
                super(ContentViewVersion, self).path(which='self'),
                which
            )
```
?

File: nailgun/entities.py (hunk 1 above)
Comment (abalakh) [inline nailgun/entities.py]: Please, rename it to `def incremental_update` to match its path.

File: nailgun/entities.py (hunk 4 above)
Comment (Ichimonji10) [inline nailgun/entities.py]: That won't work correctly. It'll produce a path like so:

    /katello/api/v2/content_view_versions/<id>/incremental_update

Which is incorrect. The current implementation will sometimes produce that path, too. `which` should be "base":

```python
super(ContentViewVersion, self).path(which='self'),
```

**Comment by sthirugn** [comment]
Rebase got messed up, fixing it

**Comment by sthirugn** [comment]
@Ichimonji10 my changes are done, please cherry pick

**Comment by sthirugn** [comment]
this pr is tested with robottelo tests, the pr for robottelo will follow soon.

**Comment by Ichimonji10** [comment]
Cherry-picked on to #200.

**Comment by sthirugn** [comment]
The commit in this pr is cherrypicked to https://github.com/SatelliteQE/nailgun/pull/200.  Closing this pr.

**Pull request closed without merging**